bot/src/functions/fetchMessages.ts

```
@@ -0,0 +1,489 @@
+            }
+        } else if (response.status === 429) {
+            const rateLimitInfo = await response.json()
+            const retryAfter = rateLimitInfo.retry_after * 1000
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            logger.warn(
+                { channel_id: channelId, message_id: messageId },
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                `Rate limited for emoji: ${encodedEmoji}. Retrying after ${rateLimitInfo.retry_after} seconds.`
+            )
+            await new Promise((resolve) => setTimeout(resolve, retryAfter))
+            continue
```

Biome

[error] Unnecessary continue statement

Unsafe fix: Delete the unnecessary continue statement


(lint/correctness/noUnnecessaryContinue)